gui/MainWindow.cxx

```diff
@@ -261,6 +261,40 @@ class StateValue : public qtUiState::AbstractItem
   T data;
 };
 
+//-----------------------------------------------------------------------------
+class RecursionGuard
```

Comment:
Should this utility be in a separate file?  It look reusable and we already have a **lot** of code in `MainWindow.cxx`

Reply:
Well, it should probably be in qtExtensions 🙂, but I didn't want to go to the effort.